skorch/utils.py

```diff
@@ -604,6 +622,15 @@ def _infer_predict_nonlinearity(net):
     if isinstance(criterion, BCEWithLogitsLoss):
         return _sigmoid_then_2d
 
+    likelihood = getattr(net, 'likelihood_', None)
+    if (
+            likelihood
+            and GPYTORCH_INSTALLED
+            and isinstance(likelihood, gpytorch.likelihoods.SoftmaxLikelihood)
```

Comment:
The code using `SoftmaxLikelihood` is commented out, so this code path is never reached?

Reply:
Indeed, good catch, I commented this code out (adding a comment), as well as the code for `_transpose`.